api/test/redis-manager.ts

```diff
@@ -2,128 +2,101 @@
 // Licensed under the MIT License.
 
 import * as assert from "assert";
-import * as express from "express";
-import * as q from "q";
 import * as shortid from "shortid";
-import Promise = q.Promise;
-
 import { RedisManager, CacheableResponse } from "../script/redis-manager";
 
 class DummyExpressResponse {
-  public statusCode: number;
-  public body: string;
-  public locals: Object;
+  public statusCode?: number;
+  public body?: string;
+  public locals: Record<string, unknown> = {};
 
   public status(statusCode: number): DummyExpressResponse {
-    assert(!this.statusCode);
+    assert(!this.statusCode, "Status code already set");
     this.statusCode = statusCode;
     return this;
   }
 
   public send(body: string): DummyExpressResponse {
-    assert(!this.body);
+    assert(!this.body, "Body already set");
     this.body = body;
     return this;
   }
 
   public reset(): void {
-    delete this.statusCode;
-    delete this.body;
+    this.statusCode = undefined;
+    this.body = undefined;
```

Comment:
do we want to set this as undefined? why?

Reply:
This is used in tests. Will evaluate it while open sourcing.